docs/deploys.md

```diff
@@ -0,0 +1,120 @@
+# Halyard Deploys
+
+## Purpose
+
+Explain how Halyard handles deploying Spinnaker.
+
+## Terminology
+
+* **Deployment Type**: How Spinnaker is divided among all the instances it is
+ running on. An example is having each Spinnaker service run a separate
+ instance. See [Deployment Types](#deployment-types) section for more
+ examples.
+* **Service Dependencies**: Dependencies of Spinnaker that need to be run as
+ stand-alone services. An example is Redis, a non-example is Packer.
+* **Deployment Environment**: The infrastructure Spinnaker is running on,
+ Spinnaker's Deployment Type, and Spinnaker's service dependencies.
+
+## Deployment Types
+
+* **LocalhostDebian**: This is every Spinnaker service installed as a Debian
+ package on the node Halyard is running on.
+* **LocalhostGit**: This is every Spinnaker service installed from Git on the 
+ node Halyard is running on. This is intended for use by Developers.
+* **Flotilla**: This is every Spinnaker service installed remotely on a
+ specified Cloud Provider as separate, independently scalable instance. There
+ is no customization or breakout of specialized profiles for this deployment.
+
+Every deployment type must address a few issues:
+
+1. **Configuration**: How Spinnaker recieves the config it needs to run.
+2. **Service Dependency Setup**: What services Spinnaker must install to get
+ this deployment to run.
+3. **Deployment & Updates**: How the Spinnaker services are deployed and
+ updated in this environment.
+
+### LocalhostDebian Deployments
+
+> Timeline: Already included in Halyard.
+
+Halyard handles this deployment type by generating an install script very
+similar to `InstallSpinnaker.sh` in spinnaker/spinnaker repository. The CLI
+receieves this script, and attempts to run it locally. 
+
+1. **Configuration**: The script generated by Halyard has commands to copy all
+ configuration from the necesssary Spinnaker config already generated by 
+ Halyard into `/opt/spinnaker/config`, `/opt/spinnaker-monitoring` and 
+ `/etc/apache2`. Any required files are already in the correct location.
+2. **Service Dependency Setup**: The only service dependency is Redis, and it
```

Comment:
If you dont need to be specific about Redis being the only one than dont. Think in terms of policy here. If someone added a new dependency, how should it behave? Provide the guidance here by speaking in terms of policies so that the person introducing something new in the future doesnt need to think about it since it is already defined.